src/list/List.tsx

```diff
@@ -0,0 +1,61 @@
+import "./_list.scss";
+
+import React, {Fragment} from "react";
+import classNames from "classnames";
+
+interface ListProps<Item = any> {
+  testid: string;
+  customClassName?: string;
+  items: Item;
+  children: (item: Item, testid: string, index?: number) => JSX.Element;
+  listItemKeyGenerator?: (item: Item, testid: string) => string;
+  placeholderProps?: {
+    shouldDisplayPlaceholder: boolean;
+    placeholder: React.ReactNode;
+  };
+  emptyStateProps?: {
+    shouldDisplayEmptyState: boolean;
+    emptyState: React.ReactNode;
+  };
+}
+
+/* eslint-disable complexity */
+function List<Item extends any>({
+  items,
+  children,
+  customClassName,
+  testid,
+  listItemKeyGenerator,
+  placeholderProps,
```

Comment:
```suggestion
  pendingStateProps,
```
Wouldn't it be better if it was named `pendingStateProps`? I cannot come up with a different use case and `emptyStateProps` is a placeholder, too.